### ISSUE TITLE
[occm] Loadbalancer Classes are not working with internal Loadbalancern

### ISSUE THREAD

**Issue by shaardie** [kubernetes/cloud-provider-openstack#1961]
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line: 
>
/kind bug
> /kind feature

**What happened**:

If setting the annotations `service.beta.kubernetes.io/openstack-internal-load-balancer: true` and `loadbalancer.openstack.org/class: someClass`, the class is ignored and the default loadbalancer settings are used.

**What you expected to happen**:

If I set a Loadbalancer Class on a Kubernetes Service, I expect it to be used. Of cource Floating IP Settings are kind of pointless, but I still want to be able to change subnet-id, etc.

**How to reproduce it**:

Create a Kubernetes Service with

```yaml
metadata:
  annotations:
    loadbalancer.openstack.org/class: someClass
    service.beta.kubernetes.io/openstack-internal-load-balancer: "true"
```

and see how the configuration in the `someClass` Loadbalancer class is going to be ignored.

**Anything else we need to know?**:

For IPv6 we are not using Floating IP, but routing the IPv6 Subnets directly into Openstack. For this szenario we use internal loadbalancer, because we do not need the Floating IPs.

**Environment**:
- openstack-cloud-controller-manager(or other related binary) version: v1.23.4
- OpenStack version:
- Others:

**Comment by jichenjc** [comment]
I didn't design the original things but reviewed your comments seems it's reasonable 
basically https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/openstack/loadbalancer.go#L2024 is the flow handle external LB and honor the class settings
I think you want to honor such at 
https://github.com/kubernetes/cloud-provider-openstack/blob/master/pkg/openstack/loadbalancer.go#L2086
as well, the question I am having is all the handle for external LB is currently floating ip central
so didn't fully get this `but I still want to be able to change subnet-id, etc.` what's the subnet-id change means here?
the ipv6 network you are using you want to select different subnet?

**Comment by shaardie** [comment]
> the question I am having is all the handle for external LB is currently floating ip central
so didn't fully get this but I still want to be able to change subnet-id, etc. what's the subnet-id change means here?

Yes, this was the next thing I realized yesterday, after looking at the code a little closer. Maybe to explain this:

We are currently using 1.20.X where it was possible to set subnet-id and network-id and other values on internal Loadbalancer via a class. This is another thing that changed during time. Which leads to another Problem that the workaround for #1086 (Workaround is described in https://github.com/kubernetes/cloud-provider-openstack/issues/1086#issuecomment-650176521) is also not working anymore. I created a new Issue for this some time ago #1855.

If this is something the project is willing to accept, we would be happy to implement it (internal Loadbalancer Classes, subnet-id and network-ids in classes, general support for setting member-ids).
